tests/manual/monkey/src/com/oracle/tools/fx/monkey/tools/KeyboardEventViewer.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (c) 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package com.oracle.tools.fx.monkey.tools;
+
+import javafx.scene.control.Button;
+import javafx.scene.control.IndexRange;
+import javafx.scene.control.TextArea;
+import javafx.scene.control.ToolBar;
+import javafx.scene.input.KeyEvent;
+import javafx.scene.layout.BorderPane;
+import com.oracle.tools.fx.monkey.util.FX;
+
+/**
+ * Keyboard Event Viewer
+ */
+public class KeyboardEventViewer extends BorderPane {
```

Comment:
Yes! Thank you for adding a keyboard event viewer. It is long overdue.

On Mac and Linux dead keys are handled using InputMethodEvents so I find it essential to catch them and dump out some minimal information. Here's the code I use (`addToLog` simply appends the text to a TextArea with a newline at the end):

```
    private void inputMethodTextChangedEvent(InputMethodEvent e) {
        if (!e.getCommitted().isEmpty()) {
            addToLog("  Commit: " + e.getCommitted());
        }
        if (!e.getComposed().isEmpty()) {
            String composedText = "";
            for (InputMethodTextRun run : e.getComposed()) {
                composedText += run.getText();
            }
            addToLog("Composed: " + composedText);
        }
    }

```

Reply:
Thank you so much - this is a very good suggestion.  Added.